Library/Homebrew/extend/os/linux/requirements/java_requirement.rb

```diff
@@ -0,0 +1,24 @@
+require "language/java"
+
+class JavaRequirement < Requirement
+  fatal true
+  default_formula "jdk"
+  download "http://www.oracle.com/technetwork/java/javase/downloads/index.html"
+
+  satisfy build_env: false do
+    jdk = Formula["jdk"]
+    @java_home = jdk.opt_prefix if jdk && jdk.installed?
+    next quiet_system "java", "-version"
```

Comment:
Given that parsing the output will likely be identical on both Linux and macOS it feels like a good time to try and fix this now, too. I'm happy to help here. Is `java_home` not available on Linux? If it is (even indirectly or in `/usr/lib`) it'll likely help you re-use the parsing code below.